### PR TITLE
make vsphere credentials public

### PR DESCRIPTION
**Pull request by xrstf**
**What this PR does / why we need it**:
The installer is planned to use this package to create the datacenters.yaml and therefore needs access to this struct as well.

To make the output somewhat prettier, I also set the optional/rarely-used fields to `omitempty`.

**Release note**:
```release-note
NONE
```

### CODE REVIEW

**Comment by xrstf** [comment]
The failed test is not caused by this change, but because of #2041.

**Review by alvaroaleman — DISMISSED** [review]

File: api/pkg/provider/datacenter.go [hunk 1]
```diff
@@ -58,7 +58,7 @@ type VSphereCredentials struct {
 // VSphereSpec describes a vsphere datacenter
 type VSphereSpec struct {
 	Endpoint      string `yaml:"endpoint"`
-	AllowInsecure bool   `yaml:"allow_insecure"`
+	AllowInsecure bool   `yaml:"allow_insecure,omitempty"`
```
Comment (alvaroaleman) [inline api/pkg/provider/datacenter.go]: Why the omitempty on non-pointers?

Comment (xrstf) [reply api/pkg/provider/datacenter.go]: Just to make it look prettier. I'm fine with removing it, the omitempty was just meant for the specs and then it kind of escalated from there :-D

Comment (alvaroaleman) [reply api/pkg/provider/datacenter.go]: I'd prefer it that way

**Review by alvaroaleman — APPROVED** [review]

**Pull request merged**